packages/asset-swapper/src/utils/market_operation_utils/constants.ts

```diff
@@ -2349,11 +2352,33 @@ export const DEFAULT_GAS_SCHEDULE: Required<FeeSchedule> = {
         return gas;
     },
     [ERC20BridgeSource.UniswapV3]: (fillData?: FillData) => {
-        let gas = 100e3;
-        const path = (fillData as UniswapV3FillData).tokenAddressPath;
-        if (path.length > 2) {
-            gas += (path.length - 2) * 32e3; // +32k for each hop.
+        const uniFillData = fillData as UniswapV3FillData | FinalUniswapV3FillData;
+        // NOTE: This base value was heuristically chosen by looking at how much it generally
+        // underestimated gas usage
+        const base = 34e3; // 34k base
+        let gas = base;
+        if (isFinalUniswapV3FillData(uniFillData)) {
+            gas += uniFillData.gasUsed;
+        } else {
+            // NOTE: We don't actually know which of the paths would be used in the router
```

Comment:
What if we just used the median? Maybe even quicker since it's already (probably) sorted?

Reply:
Great idea! We probably still need to filter out 0 values because we can end up with a bunch of trailing 0s, but inside `pathAmountsWithGasUsed` they are roughly sorted (minor price variations) 🙏